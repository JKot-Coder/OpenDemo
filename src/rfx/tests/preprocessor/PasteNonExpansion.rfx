// :RUN_RFXC -P %STD_OUTPUT% %INPUT%
//
// This os a regression test for a bug in preprocessor macro expansion.

#define CONCAT2(x, y) x ## y
#define CONCAT(x, y) CONCAT2(x, y)

#define SOMETHING someThing

// Should be someThingElse
CONCAT(SOMETHING, Else)
// Should be SOMETHINGAnother, but old preprocessor expands to produce someThingAnother
CONCAT2(SOMETHING, Another)