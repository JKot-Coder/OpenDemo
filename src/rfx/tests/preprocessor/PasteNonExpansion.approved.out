SourceOutput:[
    #line 11 "D:/Work/my/OpenDemo/src/rfx/tests/preprocessor/PasteNonExpansion.rfx"
    someThingElse
    #line 13
    SOMETHINGAnother
]
Diagnostic:[
]
